src/app/clusters/time-synchronization-server/CodegenIntegration.cpp

```diff
@@ -0,0 +1,174 @@
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::TimeSynchronization;
+using namespace chip::app::Clusters::TimeSynchronization::Attributes;
+using namespace chip::Protocols::InteractionModel;
+
+namespace {
+
+static constexpr size_t kTimeSynchronizationFixedClusterCount =
+    TimeSynchronization::StaticApplicationConfig::kFixedClusterConfig.size();
+
+static_assert((kTimeSynchronizationFixedClusterCount == 0) ||
+                  ((kTimeSynchronizationFixedClusterCount == 1) &&
+                   TimeSynchronization::StaticApplicationConfig::kFixedClusterConfig[0].endpointNumber == kRootEndpointId),
+              "Time Synchronization cluster MUST be on endpoint 0");
+
+LazyRegisteredServerCluster<TimeSynchronizationCluster> gServer;
+
+TimeSynchronization::Delegate * gDelegate = nullptr;
+
+class IntegrationDelegate : public CodegenClusterIntegration::Delegate
+{
+public:
+    ServerClusterRegistration & CreateRegistration(EndpointId endpointId, unsigned clusterInstanceIndex,
+                                                   uint32_t optionalAttributeBits, uint32_t rawFeatureMap) override
+    {
+        const BitFlags<Feature> featureMap(rawFeatureMap);
+
+        SupportsDNSResolve::TypeInfo::Type supportsDNSResolve = false;
+        if (featureMap.Has(Feature::kNTPClient))
+        {
+            VerifyOrDie(SupportsDNSResolve::Get(endpointId, &supportsDNSResolve) == Status::Success);
+        }
+
+        TimeZoneDatabaseEnum timeZoneDatabase = TimeZoneDatabaseEnum::kNone;
+        if (featureMap.Has(Feature::kTimeZone))
+        {
+            VerifyOrDie(TimeZoneDatabase::Get(endpointId, &timeZoneDatabase) == Status::Success);
+        }
+
+        TimeSynchronizationCluster::OptionalAttributeSet optionalAttributeSet(optionalAttributeBits);
+        TimeSourceEnum timeSource = TimeSourceEnum::kNone;
+        if (optionalAttributeSet.IsSet(TimeSource::Id))
+        {
+            VerifyOrDie(TimeSource::Get(endpointId, &timeSource) == Status::Success);
+        }
+
+        NTPServerAvailable::TypeInfo::Type ntpServerAvailable = false;
+        if (featureMap.Has(Feature::kNTPServer))
+        {
+            VerifyOrDie(SupportsDNSResolve::Get(endpointId, &ntpServerAvailable) == Status::Success);
+        }
+
+        gServer.Create(endpointId, featureMap, optionalAttributeSet,
+                       TimeSynchronizationCluster::StartupConfiguration{ .supportsDNSResolve = supportsDNSResolve,
+                                                                         .ntpServerAvailable = ntpServerAvailable,
+                                                                         .timeZoneDatabase   = timeZoneDatabase,
+                                                                         .timeSource         = timeSource,
+                                                                         .delegate           = gDelegate });
```

Comment:
gDelegate is set by the GetDelegate (which is unintuitive, but ok ...), so I believe we should use GetDelegate here (I don't think we loop since gServer would not be created yet). This is to avoid a null delegate.